### DISCUSSION TITLE
[HELP] integrating SSELink and httpLink

### DISCUSSION THREAD

**Discussion by ebo-bpi** [enisdenjo/graphql-sse#89] [Q&A]
I'm trying to integrate SSE with my normal apollo connection to be able to use it with query and subscription.
I've setup the links as follows:

```
class SSELink extends ApolloLink {
  private client: Client;

  constructor(options: ClientOptions) {
    super();
    this.client = createClient(options);
  }

  public request(operation: Operation): Observable<FetchResult> {
    return new Observable((sink) =>
      this.client.subscribe<FetchResult>(
        { ...operation, query: print(operation.query) },
        {
          next: sink.next.bind(sink),
          complete: sink.complete.bind(sink),
          error: sink.error.bind(sink),
        }
      )
    );
  }
}

const httpLink = new HttpLink({
  uri: "/query",
  credentials: "include",
});
const sselink = new SSELink({
  url: "/query",
  credentials: "include",
});

const splitLink = split(
  ({ query }) => {
    const definition = getMainDefinition(query);
    return (
      definition.kind === "OperationDefinition" &&
      definition.operation === "subscription"
    );
  },
  sselink,
  httpLink
);

const createApolloClient = () =>
  new ApolloClient({
    ssrMode: typeof window === "undefined",
    cache: new InMemoryCache(),
    link: splitLink,
    credentials: "include",
  });
```

Then I use this client with ApolloProvider and pass this client on. 
For the httpLink all is working fine. 
For the sseLink it is setting up the call to the backend, 
but I'm getting an error in my browser after some time: NS_ERROR_NET_PARTIAL_TRANSFER

Couldn't find anything useful on this error. 
Question I have: Can we use mix the links the way I've set it up?

**Comment by enisdenjo** [comment]
NS_ERROR_NET_PARTIAL_TRANSFER usually means that the size of the response does not match the `Content-Length` header; however, this is weird because the `Content-Length` header should not be a part of an SSE connection (since the response size is not known ahead of time).

Are you sure you're getting this error with SSE? What is the server implementation you're using?
